rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -34,33 +36,67 @@
     // default values
     var state = {
       is_open: false,
-      is_staff: false,
       runs: [],
       choices: [],
       counts: {},
-      selectedRuns: [null]  // one per chart. null means select the latest one
+      selectedRuns: [null],  // one per chart. null means select the latest one
+      isPolling: false,  // is there a request in progress? Used to disable button to prevent double clicks
+      lastPoll: null  // a moment object representing the time at last poll, to be used to diff with the run
     };
 
     /**
      * Render template
      */
     function render() {
       var $rapidBlockContent = $element.find(rapidBlockContentSel);
-      $rapidBlockContent.html(toggleTemplate(state));
+      var pollSeconds = 0, pollMinutes = 0;
+      if (state.is_open) {
+        var openRun = state.runs.find(function(run) {
+          return run.open;
+        });
+
+        var totalSeconds = 0;
+        // It should almost always be true that if state.is_open is true that openRun exists
+        // But there is a small delay between when state.is_open is set and when the runs
+        // are refreshed from the server
+        if (openRun) {
+
+          // millis is the time between current time and the time of last fetch (according to browser), plus
+          // the time between the last fetch and the run creation (according to server)
+          var millis = moment().diff(state.lastPoll) + moment(state.server_now).diff(moment(openRun.created));
+          totalSeconds = Math.floor(millis / 1000);
```

Comment:
If/when we change the rendering approach to use jQuery to do the rendering, it would probably make sense to (a) move this calculation logic into a helper method, and (b) only call it on the initial page load (i.e.: let the `setTimeout` update the counts without looking at the time difference between now and the open run created date)

Reply:
I think `setTimeout` would drift out of sync with the calculated time, especially if the tab is in the background.